src/language-js/print/assignment.js

```diff
@@ -67,18 +67,26 @@ function printAssignmentRight(leftNode, rightNode, printedRight, options) {
     return indent([line, printedRight]);
   }
 
-  /**
-   * @param {Node} node
-   * @returns {Node}
-   */
-  function getUnaryArgument(node) {
-    if (node.type === "UnaryExpression") {
-      return getUnaryArgument(node.argument);
-    }
-    return node;
+  if (canBreakAssignmentRight(leftNode, rightNode, options)) {
+    return group(indent([line, printedRight]));
   }
 
-  const canBreak =
+  return [" ", printedRight];
+}
+
+/**
+ * @param {Node} node
+ * @returns {Node}
+ */
+function getUnaryArgument(node) {
```

Comment:
We can use a loop here, I'm not sure if we should only check `!`, `!+a` `+!a` seems very uncommon.

Reply:
> We can use a loop here.

Sounds good.

>  `!+a` `+!a` seems very uncommon.

I think so too. But I don't think it's necessary to add a condition to exclude it.

Reply:
If you agree it's not common, I think we should add one more condition, simple

```js
const node =
  node.type === "UnaryExpression" &&
  node.operator === "!" &&
  node.argument.type === "UnaryExpression" &&
  node.argument.operator === "!"
    ? node.argument.argument
    : node.argument;
```

Only any `UnaryOperator` + `!!`

Reply:
This function confused me... I thought it's always `UnaryExpression`, I wounder why I keep failing on write these checks... It also allow `StringLiteral`

Reply:
I'm done with rewriting `canBreakAssignmentRight`, I think it is more clear now.

I removed `getUnaryArgument` function, as it's just one line, and this function name is confusing. 

I'll let you decide if accept my suggestion.